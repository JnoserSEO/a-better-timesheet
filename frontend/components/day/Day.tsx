import React from 'react'
import { Entries, EntriesProps } from '../entry/Entries'
import '../../styles/day/day.scss'
import '../../styles/utility/utility.scss'

const Day:React.FC<EntriesProps> = ({entries})=>{
    const hours = entries.reduce((acc, curr) => acc + curr.hours, 0)
    return <div className="day-inner">
        <Entries entries={entries} />
        <div className="add-div"><span className='add'>+</span></div>
        <div className="day-total">{`${hours} Hour${hours > 1 ? 's':''}`}</div>
    </div>
}

export default Day