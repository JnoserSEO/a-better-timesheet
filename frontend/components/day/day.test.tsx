import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Day from './Day'


beforeEach(() => {
    beforeEach(() => {
        Date.now = jest.fn(() => 1601401562455); // 2020-9-29
    });
});

test('should render Day correctly with no entries',()=>{
    const wrapper = shallow(<Day date={new Date(1601401562455)} entries={[]} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})
