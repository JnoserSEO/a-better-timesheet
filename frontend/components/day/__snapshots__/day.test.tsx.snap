// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Day correctly with no entries 1`] = `
<td
  className="hours-td"
>
  <Component
    dayOfWeek="Tuesday"
    entries={Array []}
  />
</td>
`;
